adapter.go

```diff
@@ -296,11 +297,33 @@ func (a *Adapter) casbinRuleTable() func(db *gorm.DB) *gorm.DB {
 
 func (a *Adapter) createTable() error {
 	t := a.db.Statement.Context.Value(customTableKey{})
+
+	tableName := a.tableName
+	if a.tablePrefix != "" {
+		tableName = a.tablePrefix + "_" + tableName
+	}
+	index := "idx_" + tableName
+
+	hasIndex := false
 	if t == nil {
-		return a.db.AutoMigrate(a.getTableInstance())
+		instance := a.getTableInstance()
+		hasIndex = a.db.Migrator().HasIndex(instance, index)
+		if err := a.db.AutoMigrate(instance); err != nil {
+			return err
+		}
+	} else {
+		hasIndex = a.db.Migrator().HasIndex(t, index)
+		if err := a.db.AutoMigrate(t); err != nil {
+			return err
+		}
 	}
```

Comment:
this can be simplified 
```go
if t == nil {
		t = a.getTableInstance()
		hasIndex = a.db.Migrator().HasIndex(instance, index)
	} else {
		hasIndex = a.db.Migrator().HasIndex(t, index)
		
        }
        if err := a.db.AutoMigrate(t); err != nil {
			return err
	}
```